src/safeds/ml/nn/_model.py

```diff
@@ -57,33 +71,39 @@ def fit(
             raise OutOfBoundsError(actual=epoch_size, name="epoch_size", lower_bound=ClosedBound(1))
         if batch_size < 1:
             raise OutOfBoundsError(actual=batch_size, name="batch_size", lower_bound=ClosedBound(1))
+        if train_data.features.number_of_columns is not self._input_size:
+            raise InputSizeError(train_data.features.number_of_columns, self._input_size)
+        self._feature_names = train_data.features.column_names
+        self._batch_size = batch_size
```

Comment:
This should only be set on the copied model so that we don't change the current model in-place